docs/source/hooks/common_use_cases.md

```diff
@@ -200,3 +200,22 @@ HOOKS = (AzureSecretsHook(),)
 ```{note}
 Note: `DefaultAzureCredential()` is Azure's recommended approach to authorise access to data in your storage accounts. For more information, consult the [documentation about how to authenticate to Azure and authorize access to blob data](https://learn.microsoft.com/en-us/azure/storage/blobs/storage-quickstart-blobs-python).
 ```
+
+## Use Hook to read `metadata` from `DataCatalog`
+We recommend to use the `after_catalog_created` hook if you need to access `metadata` to extend Kedro.
+
+```python
+class MetadataHook:
+    @hook_impl
+    def after_catalog_created(
+        self,
+        catalog: DataCatalog,
+        conf_catalog: Dict[str, Any],
+        conf_creds: Dict[str, Any],
+        feed_dict: Dict[str, Any],
+        save_version: str,
+        load_versions: Dict[str, str],
```

Comment:
All these parameters are not used right? Maybe we can remove them from the function. Pluggy allows this

Reply:
Fair point. We do this inconsistently, sometimes we do include signature that wasn't used (I was asked to add that before, can't find it exactly which PR it is). 

For example, `after_node_run` has a redundant `node` argument.
https://docs.kedro.org/en/stable/hooks/examples.html#add-observability-to-your-pipeline

Reply:
@stichbury do you think we should keep all our example minimals? We can do a clean up if this is agreed.

In this case
```
   def after_catalog_created(
        self,
        catalog: DataCatalog,
        conf_catalog: Dict[str, Any],
        conf_creds: Dict[str, Any],
        feed_dict: Dict[str, Any],
        save_version: str,
        load_versions: Dict[str, str])
```

or 
```
   def after_catalog_created(
        self,
        catalog: DataCatalog)
```

will both work.